bach/tests/conftest.py

```diff
@@ -43,26 +45,54 @@ def pytest_addoption(parser: Parser):
     parser.addoption('--all', action='store_true', help='run the functional tests for BigQuery')
 
 
+class EngineDialect(NamedTuple):
+    engine: Optional[Engine]
+    dialect: Optional[Dialect]
+
+
+ENGINE_DIALECTS: Dict[str, EngineDialect] = {}
+
+
+def pytest_sessionstart(session: Session):
+    # Initialize ENGINE_DIALECTS per pytest-xdist worker session based on current pytest running config.
+    # This way we avoid the creation of a new engine per test, as this is quite inefficient and might
+    # cause failures due to multiple clients trying to connect to db server.
+
+    # For more information, please see:
+    # https://pytest-xdist.readthedocs.io/en/latest/distribution.html
+    # https://docs.pytest.org/en/6.2.x/reference.html#pytest.hookspec.pytest_sessionstart
+    load_engine = True
+    if (
+        ('tests/unit' in session.config.args and len(session.config.args) == 1)
+        or session.config.getoption('markexpr') == 'db_independent'
+    ):
+        # create dialect only if we are running unit tests or db independent tests
```

Comment:
```suggestion
        # create only dialect, if we are running unit tests or db independent tests
```

Reply:
Maybe even
```suggestion
        # create only dialect, if we are running unit tests
```

As the db-independent marker just implies that it is a unit-test.